datasets/bnl_processed_newspapers/bnl_processed_newspapers.py

```diff
@@ -0,0 +1,186 @@
+                        "TABLE",
+                        "ADVERTISEMENT",
+                        "CHART_DIAGRAM",
+                        "ILLUSTRATION",
+                        "ISSUE",
+                    ]
+                ),
+                "extent": datasets.Value("int32"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "dirpath": os.path.join(data_dir),
+                },
+            ),
+        ]
+
+    def _generate_examples(
+        self,
+        dirpath,
+    ):
+        """Yields examples as (key, example) tuples."""
+        for id_, xml in enumerate(Path(dirpath).rglob("**/*.xml")):
+            tree = ET.parse(xml)
+            root = tree.getroot()
+            source = tree.findall(".//{http://purl.org/dc/elements/1.1/}source")[0].text
+            metadata = root.find(".//{http://www.openarchives.org/OAI/2.0/}metadata")
+            ispartof = metadata.find(".//{http://purl.org/dc/terms/}isPartOf").text
+            date = metadata.find(".//{http://purl.org/dc/elements/1.1/}date").text
+            if date:
+                date = datetime.strptime(date, "%Y-%m-%d")
+            publisher = root.findall(".//{http://purl.org/dc/elements/1.1/}publisher")
+            if publisher:
+                publisher = publisher[0].text
+            hasversion = metadata.find(".//{http://purl.org/dc/terms/}hasVersion").text
+
+            description = tree.findall(".//{http://purl.org/dc/elements/1.1/}description")[0].text
+            ark_id = tree.findall(".//{http://purl.org/dc/elements/1.1/}identifier")[0].text
+            title = metadata.find(".//{http://purl.org/dc/elements/1.1/}title").text
+            article_type = metadata.find(".//{http://purl.org/dc/elements/1.1/}type").text
+            extent = metadata.find(".//{http://purl.org/dc/terms/}extent").text
+
```

Comment:
Maybe you missed the "language" tag? I think this information is very important, because I think they have articles in French and German (so that users could filter on this).

Reply:
Yes, this is definetly an ommision. I have added this in now.